test/tests/functional/pbs_ralter.py

```diff
@@ -1108,3 +1109,153 @@ def test_conflict_standing_resv_occurrence(self):
 
         self.alter_a_reservation(rid, start, end, shift, alter_e=True,
                                  whichMessage=0)
+
+    def test_alter_advance_resv_boundary_values(self):
+        """
+        This test checks the alter of start and end times at the boundary
+        values for advance reservation
+        """
+        duration = 30
+        shift = 5
+        offset = 5
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=4")
+
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_e=True, sequence=1)
+        start, end = self.alter_a_reservation(
+            rid, start, end, -shift, alter_e=True, sequence=2)
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_s=True, sequence=3)
+        self.alter_a_reservation(
+            rid, start, end, -shift, alter_s=True, sequence=4)
+
+    def test_alter_standing_resv_boundary_values(self):
+        """
+        This test checks the alter of start and end times at the boundary
+        values for standing reservation
+        """
+        duration = 30
+        shift = 5
+        offset = 5
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=4", standing=True,
+            rrule="FREQ=DAILY;COUNT=10")
+
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_e=True, sequence=1)
+        start, end = self.alter_a_reservation(
+            rid, start, end, -shift, alter_e=True, sequence=2)
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_s=True, sequence=3)
+        self.alter_a_reservation(
+            rid, start, end, -shift, alter_s=True, sequence=4)
+
+    def test_alter_degraded_resv_mom_down(self):
+        """
+        This test checks the alter of start and end times of reservations
+        when mom is down
+        """
+        duration = 30
+        shift = 5
+        offset = 5
+
+        rid1, start1, end1 = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=2")
+        rid2, start2, end2 = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=2", standing=True,
+            rrule="FREQ=DAILY;COUNT=10")
+        self.mom.stop()
+        msg = 'mom is not down'
+        self.assertFalse(self.mom.isUp(), msg)
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
+        self.server.expect(RESV, attrs, id=rid1, max_attempts=30, interval=0.5)
+        self.server.expect(RESV, attrs, id=rid2, max_attempts=30, interval=0.5)
```

Comment:
The default interval is 0.5.  You can remove it.  The default max_attempts is 60, so consider using that.

Reply:
Done